epsagon_kubernetes_subscription.sh

```diff
@@ -66,30 +67,49 @@ function send_to_epsagon {
 function apply_role {
     EPSAGON_TOKEN=$1
     KUBECTL="kubectl"
-    if [ ! -z $2 ] ; then
-        CONTEXT=$2
+    CONFIG=$2
+    if [ ! -z $3 ] ; then
+        CONTEXT=$3
         KUBECTL="kubectl --context ${CONTEXT}"
         echo "Applying ${ROLE_FILE} to ${CONTEXT}"
     else
         echo "Applying ${ROLE_FILE}"
     fi
     echo ""
-    ${KUBECTL} apply -f ${ROLE_FILE}
-
-    SA_SECRET_NAME=`${KUBECTL} -n epsagon-monitoring get secrets | grep 'epsagon-monitoring-token' | awk '{print $1}'`
+    ${KUBECTL} apply -f ${ROLE_FILE} --kubeconfig=${CONFIG}
+    SA_SECRET_NAME=`${KUBECTL} -n epsagon-monitoring get secrets --kubeconfig=${CONFIG} | grep 'epsagon-monitoring-token' | awk '{print $1}'`
     if [ `which python` ] ; then
-        ROLE_TOKEN=`${KUBECTL} -n epsagon-monitoring get secrets $SA_SECRET_NAME -o json | python -c 'import sys, json; print(json.load(sys.stdin)["data"]["token"])' | base64 --decode`
+        ROLE_TOKEN=`${KUBECTL} -n epsagon-monitoring get secrets $SA_SECRET_NAME -o json --kubeconfig=${CONFIG} | python -c 'import sys, json; print(json.load(sys.stdin)["data"]["token"])' | base64 --decode`
     else
-        ROLE_TOKEN=`${KUBECTL} -n epsagon-monitoring get secrets $SA_SECRET_NAME -o json | grep '\"token\"' | cut -d: -f2 | cut -d'"' -f2 | base64 --decode`
+        ROLE_TOKEN=`${KUBECTL} -n epsagon-monitoring get secrets $SA_SECRET_NAME -o json --kubeconfig=${CONFIG} | grep '\"token\"' | cut -d: -f2 | cut -d'"' -f2 | base64 --decode`
     fi
 
-    send_to_epsagon $EPSAGON_TOKEN $ROLE_TOKEN $CONTEXT
+    send_to_epsagon $EPSAGON_TOKEN $ROLE_TOKEN $CONFIG $CONTEXT
+
+}
 
+function does_config_file_exist {
+    if [ -f ~/.kube/config ]; then
+        return 0
+    fi
+    IFS=';' read -ra ADDR <<< $KUBECONFIG
+    for i in ${ADDR[@]}; do
```

Comment:
I am not sure if this is well supported in zsh, will check later. also are you sure the separator is not ':'?
https://kubernetes.io/docs/tasks/access-application-cluster/configure-access-multiple-clusters/#linux-1

```suggestion
    for i in `echo $KUBECONFIG | tr ':' '\n'`; do
```